cpp/include/cudf/table/table_view.hpp

```diff
@@ -78,7 +79,17 @@ class table_view_base {
    *
    * @param cols The vector of columns to construct the table from
    */
-  explicit table_view_base(std::vector<ColumnView> const& cols);
+  explicit table_view_base(std::vector<ColumnView> const& cols) : _columns{cols}
+  {
+    if (num_columns() > 0) {
+      std::for_each(_columns.begin(), _columns.end(), [this](ColumnView col) {
+        CUDF_EXPECTS(col.size() == _columns.front().size(), "Column size mismatch.");
+      });
+      _num_rows = _columns.front().size();
+    } else {
+      _num_rows = 0;
+    }
+  }
```

Comment:
question: could we separate declarations and definitions into different files?

Reply:
Yes we can. This was originally written when the goal was the even installed `detail` types would be hidden. To work around that I inlined the constructor so that consumers could construct `table_view_base`.

Now that we are going to make this class public I can remove this change.

Reply:
I'm happy to approve once this change is reverted.

Reply:
Reverted the change